### PR TITLE
Feature/hbd 6116

### PR DESCRIPTION
**Pull request by mprzypasniak99**

### CODE REVIEW

**Review by BakJakub — CHANGES_REQUESTED** [review]
In general, the whole implementation looks correct, I just wonder whether it is not worth adding dependency injection and whether to create a Processor Register + Processor Factory

`class ProcessorRegistry {
    private var registry = [String: WebSocketProcessor.Type]()

    func registerProcessor(_ type: WebSocketProcessor.Type, for event: String) {
        registry[event] = type
    }

    func getProcessor(for event: String) -> WebSocketProcessor? {
        guard let processorType = registry[event] else { return nil }
        return processorType.init()
    }
}
`

`class ProcessorFactory {
    private let registry: ProcessorRegistry

    init(registry: ProcessorRegistry) {
        self.registry = registry
    }

    func build(event: String) -> WebSocketProcessor? {
        return registry.getProcessor(for: event)
    }
}
`

`let processorRegistry = ProcessorRegistry()
processorRegistry.registerProcessor(UserProcessor.self, for: "event.user")`

Dependency
`
class HandbidEventDelegate: EventDelegate { 
private let processorFactory: ProcessorFactory
    private let logger: Logger

    init(processorFactory: ProcessorFactory, logger: Logger = Logger()) {
        self.processorFactory = processorFactory
        self.logger = logger
    }`
    
The proposed solution seems better to me because it reduces the number of changes to the code when adding new events, in addition, the injection of dependencies also seems to me to be a better solution than the Singelton pattern, as proposed earlier

File: Handbid_iPad/Data/WebSocket/HandbidEventDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+			if changed {
+				reconnectSocket(client: client)
+			}
+		case let .reconnectSuggested(suggested):
+			if suggested {
+				reconnectSocket(client: client)
+			}
+		case .cancelled:
+			reconnectSocket(client: client)
+		case .peerClosed:
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/Data/WebSocket/HandbidEventDelegate.swift]: Fixed

File: Handbid_iPad/Data/WebSocket/HandbidEventDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			heartbeat(client: client)
+		case HandbidEventDelegate.registerMessage:
+			registerToUserChannel(client: client)
+			registerToAuctionChannel(client: client)
+		default:
+			parseEvent(message: message)
+		}
+	}
+
+	private func parseEvent(message: String) {
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/Data/WebSocket/HandbidEventDelegate.swift:82]: Thanks for the suggestion. Changed and definitely looks better now

File: Handbid_iPad/Data/WebSocket/WebSocketManager.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2024 by Handbid. All rights reserved.
+
+import NetworkService
+import Starscream
+
+class WebSocketManager {
+	static var socket: WebSocket?
+	static var delegate: EventDelegate?
+
+	static func startSocket(urlFactory: WebSocketURLFactory, token: TokenUser?) {
```
[the remark this answers is not in the document]
Comment (mprzypasniak99) [reply Handbid_iPad/Data/WebSocket/WebSocketManager.swift]: Changed to signleton pattern

**Pull request merged**